src/OpenConext/UserLifecycle/Infrastructure/UserLifecycleBundle/Command/DeprovisionCommand.php

```diff
@@ -0,0 +1,106 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\UserLifecycle\Infrastructure\UserLifecycleBundle\Command;
+
+use InvalidArgumentException;
+use OpenConext\UserLifecycle\Domain\Service\DeprovisionServiceInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+class DeprovisionCommand extends Command
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var DeprovisionServiceInterface
+     */
+    private $service;
+
+
+    public function __construct(DeprovisionServiceInterface $service, LoggerInterface $logger)
+    {
+        parent::__construct(null);
+        $this->service = $service;
+        $this->logger = $logger;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('user-lifecycle:deprovision')
+            ->setDescription('Deprovision a user from the platform. The user is identified by a collabPersonId.')
+            ->setHelp(
+                'This command allows you to deprovision a given user identified by a collabPersonId. '.
+                'The command will delegate the deprovisioning to all registered applications and report back '.
+                'on the actually removed data.'
+            )
+            ->addArgument(
+                'user',
+                InputArgument::REQUIRED,
+                'The collabPersonId of the user to deprovision.'
+            )
+            ->addOption(
+                'force',
+                'f'
+            )
+            ->addOption(
+                'dry-run',
+                null,
+                InputOption::VALUE_NONE,
+                'Should the command be run in dry run mode?'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $userIdInput = $input->getArgument('user');
+        $dryRun = $input->hasOption('dryRun');
+        $forced = $input->getOption('force');
+
+        if (!$forced) {
+            $helper = $this->getHelper('question');
+            $question = new ConfirmationQuestion(sprintf('Continue with deprovisioning of "%s"? (y/n)', $userIdInput), false);
+
+            if (!$helper->ask($input, $output, $question)) {
+                return;
+            }
+        }
+
+        $this->logger->info(
+            sprintf(
+                'Received a deprovision request for user: "%s", with dryRun turned %s',
+                $userIdInput,
+                ($dryRun ? 'on' : 'off')
+            )
+        );
+        try {
+            $output->write($this->service->deprovision($userIdInput, $dryRun));
```

Comment:
Some thoughts:

 * don't render the JSON as returned by the API, use `JSON_PRETTY_PRINT`
 * wouldn't it be more readable if we iterate over the collection and render a heading "Results from EngineBlock:" and show the JSON output indented below the heading?